### PR TITLE
Add reset room clean command

### PR DESCRIPTION
**Pull request by bryanhce**
Also updated the Developer Guide

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -192,12 +198,13 @@ _Details coming soon ..._
...
+|--------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| **Add**            | `add n/NAME p/PHONE_NUMBER e/EMAIL dr/DATE_RANGE ng/NUMBER_OF_GUESTS rc/IS_ROOM_CLEAN ​` <br> e.g., `add n/John Doe p/98765432 e/johnd@example.com dr/19/05/20 - 24/05/22 ng/3 rc/no`     |
+| **Clear**          | `clear`                                                                                                                                                                                   |
+| **Delete**         | `delete INDEX`<br> e.g., `delete 3`                                                                                                                                                       |
+| **Edit**           | `edit INDEX [n/NAME] [p/PHONE] [e/EMAIL] [dr/DATE_RANGE] [ng/NUMBER_OF_GUESTS]​`<br> e.g.,`edit 2 e/jameslee@example.com dr/13/09/22 - 15/09/22 ng/5`                                     |
```
Comment (jehousoh) [inline docs/UserGuide.md:206]: Should there be [rc/IS_ROOM_CLEAN] here?

File: src/test/java/seedu/guest/logic/commands/ResetRoomCleanCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+import seedu.guest.model.Model;
+import seedu.guest.model.ModelManager;
+import seedu.guest.model.UserPrefs;
+import seedu.guest.model.guest.Guest;
+import seedu.guest.testutil.GuestBuilder;
+
+
+
+
```
Comment (jehousoh) [inline src/test/java/seedu/guest/logic/commands/ResetRoomCleanCommandTest.java:19]: Is there any reason for these extra newlines?

File: src/main/java/seedu/guest/logic/commands/ResetRoomCleanCommand.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Guest> lastShownList = model.getFilteredGuestList();
+
+        Guest guestToEdit;
+        Guest editedGuest;
+
+        for (int index = 0; index < lastShownList.size(); index++) {
```
Comment (jehousoh) [inline src/main/java/seedu/guest/logic/commands/ResetRoomCleanCommand.java:40]: Does this only reset the lastShownList? (ie not the entire list but only the list that is searched for) (eg. Find Alex -> resetRoomClean)

**Review by jehousoh — CHANGES_REQUESTED** [review]
Nice work but I had some questions regarding the implementations! Also you might need to re-run the CI test.

File: docs/UserGuide.md (hunk 1 above)
Comment (bryanhce) [reply docs/UserGuide.md:206]: Indeed!

File: src/test/java/seedu/guest/logic/commands/ResetRoomCleanCommandTest.java (hunk 2 above)
Comment (bryanhce) [reply src/test/java/seedu/guest/logic/commands/ResetRoomCleanCommandTest.java:19]: No reason, will remove.

File: src/main/java/seedu/guest/logic/commands/ResetRoomCleanCommand.java (hunk 3 above)
Comment (bryanhce) [reply src/main/java/seedu/guest/logic/commands/ResetRoomCleanCommand.java:40]: Yes only last shown list. I thought I should standard how edits are made so that it's more intuitive for users. Currently, delete and edit commands also only work on last shown list. But I also understand that it make sense to change all the guests at once.

File: src/main/java/seedu/guest/logic/commands/ResetRoomCleanCommand.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+/**
+ * Changes isRoomClean field of all guest in current list to "no".
+ */
+public class ResetRoomCleanCommand extends Command {
+
+    public static final String COMMAND_WORD = "resetRoomClean";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Changes all guest RoomClean status in current list to \"no\".\n"
```
Comment (JustinPeng13) [inline src/main/java/seedu/guest/logic/commands/ResetRoomCleanCommand.java:27]: Could we rephrase this to "Changes all guests' RoomClean statuses"?

File: src/main/java/seedu/guest/logic/commands/ResetRoomCleanCommand.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+            + "Example: " + COMMAND_WORD;
+
+    public static final String MESSAGE_SUCCESS = "Reset all guests' Room Clean status in current list to \"no\".";
```
Comment (JustinPeng13) [inline src/main/java/seedu/guest/logic/commands/ResetRoomCleanCommand.java:30]: Similarly, could we use "RoomClean statuses"?

File: src/test/java/seedu/guest/logic/commands/ResetRoomCleanCommandTest.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+        Guest guestInFilteredList5 = model.getFilteredGuestList().get(5);
+        Guest guestInFilteredList6 = model.getFilteredGuestList().get(6);
+
+        Guest editedGuest0 = new GuestBuilder(guestInFilteredList0).withIsRoomClean(VALID_IS_ROOM_CLEAN_BOB).build();
+        Guest editedGuest1 = new GuestBuilder(guestInFilteredList1).withIsRoomClean(VALID_IS_ROOM_CLEAN_BOB).build();
+        Guest editedGuest2 = new GuestBuilder(guestInFilteredList2).withIsRoomClean(VALID_IS_ROOM_CLEAN_BOB).build();
+        Guest editedGuest3 = new GuestBuilder(guestInFilteredList3).withIsRoomClean(VALID_IS_ROOM_CLEAN_BOB).build();
+        Guest editedGuest4 = new GuestBuilder(guestInFilteredList4).withIsRoomClean(VALID_IS_ROOM_CLEAN_BOB).build();
+        Guest editedGuest5 = new GuestBuilder(guestInFilteredList5).withIsRoomClean(VALID_IS_ROOM_CLEAN_BOB).build();
+        Guest editedGuest6 = new GuestBuilder(guestInFilteredList6).withIsRoomClean(VALID_IS_ROOM_CLEAN_BOB).build();
```
Comment (JustinPeng13) [inline src/test/java/seedu/guest/logic/commands/ResetRoomCleanCommandTest.java:43]: Should we have some cleaned and uncleaned rooms to test that the command works for both?

**Review by JustinPeng13 — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/guest/logic/commands/ResetRoomCleanCommand.java (hunk 4 above)
Comment (bryanhce) [reply src/main/java/seedu/guest/logic/commands/ResetRoomCleanCommand.java:27]: Yes!

File: src/main/java/seedu/guest/logic/commands/ResetRoomCleanCommand.java (hunk 5 above)
Comment (bryanhce) [reply src/main/java/seedu/guest/logic/commands/ResetRoomCleanCommand.java:30]: Yes!

File: src/test/java/seedu/guest/logic/commands/ResetRoomCleanCommandTest.java (hunk 6 above)
Comment (bryanhce) [reply src/test/java/seedu/guest/logic/commands/ResetRoomCleanCommandTest.java:43]: This part is to construct to the expectedModel. The model to be tested has both cleaned and uncleaned rooms.

File: src/main/java/seedu/guest/logic/commands/ResetRoomCleanCommand.java (hunk 3 above)
Comment (bryanhce) [reply src/main/java/seedu/guest/logic/commands/ResetRoomCleanCommand.java:40]: I have discussed with Justin and will make the necessary edits to edit the whole list instead! Thanks for bringing it up!

**Comment by bryanhce** [comment]
This pull request is no longer in use. A new pull request has been made as the name of the branch has been changes.
